internal/apimeta/mapper.go

```diff
@@ -0,0 +1,81 @@
+package apimeta
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+// ResourceMapper allows callers to map resources to kinds.
+type ResourceMapper interface {
+	// NamespacedKindFor returns the priority mapping GVK for a namespaced GR.
+	// If the GR doesn't exist, or isn't namespaced, returns an error.
+	NamespacedKindFor(gr schema.GroupResource) (schema.GroupVersionKind, error)
+}
+
+// NotNamespacedError is returned if the mapper can find the resource, but it is not namespaced.
+type NotNamespacedError struct {
+	GroupResource schema.GroupResource
+}
+
+func (e *NotNamespacedError) Error() string {
+	return fmt.Sprintf("resource %v is not namespaced", e.GroupResource)
```

Comment:
I think I'd prefer `%q` here to `%v` to maintain the double quotes. We switched a bunch of places to `%q` throughout HNC as we found it made the errors easier to read.

Reply:
Thx, fixed!